node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/coredump/CoredumpHandler.java

```diff
@@ -151,10 +158,12 @@ Optional<ContainerPath> enqueueCoredump(ContainerPath containerCrashPath, Contai
 
     void processAndReportSingleCoredump(NodeAgentContext context, ContainerPath coredumpDirectory, Supplier<Map<String, Object>> nodeAttributesSupplier) {
         try {
-            String metadata = getMetadata(context, coredumpDirectory, nodeAttributesSupplier);
+            Optional<SecretSharedKey> sharedCoreKey = Optional.ofNullable(secretSharedKeySupplier.get());
+            Optional<String> decryptionToken = sharedCoreKey.map(k -> k.sealedSharedKey().toTokenString());
+            String metadata = getMetadata(context, coredumpDirectory, nodeAttributesSupplier, decryptionToken);
```

Comment:
`getMetadata()` writes the resulting metadata to a file, but if the file already exists, it'll return the data from the file ignoring the arguments. If we fail to report this coredump, we will try again later, but then we'll end up encrypting the core with the new key while the metadata will have the old `decryption_token`.

Reply:
No problem overwriting it, this is done to avoid having to reprocess the coredump with `gdb` etc.

Reply:
Understood. Presumably this should only happen if either a) core dump reporting fails, or b) writing the compressed/encrypted core dump fails? I'd expect both of those to be fairly rare occurrences. For now I think the pragmatic route with overwriting should be fine; we can later consider other options such as patching in a new token into the existing metadata if we feel it makes sense (it's "just JSON"™ after all).

Reply:
Agree :+1:

Reply:
Patching in the new token value turned out to be not that much extra work after all, so I went with that. PTAL.